File: python/ABSTRACT_FACTORY_DECORATOR/ABSTRACT_FACTORY_DECORATOR/reservacion/ReservacionVIP.py
from ABSTRACT_FACTORY_DECORATOR.reservacion.Reservacion import Reservacion
from .DatosBeneficios import DatosBeneficios

class ReservacionVIP(Reservacion):
    
    datosBeneficios = DatosBeneficios()
    
    def __init__(self, datos):
        super().__init__(datos[:-1])
        self.datosBeneficios = DatosBeneficios()
        self.datosBeneficios.setDatosBeneficios(datos[-1])
    
    def mostrarReservacion(self):
        super().mostrarReservacion()
        print(f"Beneficios   : {self.datosBeneficios.beneficios}")

File: php/ABSTRACT_FACTORY_DECORATOR/reservacion/ReservacionVIP.php
<?php

require_once 'Reservacion.php';
require_once 'DatosBeneficios.php';
require_once 'DatosCliente.php';
require_once 'DatosCompra.php';
require_once 'DatosFuncion.php';
class ReservacionVIP extends Reservacion{
    
    public $datosBeneficio;
    
    public function __construct($datos, $datosCliente, $datosCompra, $datosFuncion) {
        
        parent::__construct($datosCliente, $datosCompra, $datosFuncion);
        
        $this->datosBeneficio = new DatosBeneficios();
        $this->datosBeneficio->setDatosBeneficios($datos);
    }
    
    public function mostrarReservacion() {
        return parent::mostrarReservacion(). $this->datosBeneficio->getDatosBeneficios();
    }
    
    
}
